### ISSUE TITLE
Sync support

### ISSUE THREAD

**Issue by jawsh207** [PhilKes/NotallyX#164]
Enabled the option to store data in "android>media>com.philkes.notallyx" so that I could sync between PC via Syncthing. However, I don't seem to be able to view anything regardless of the note program or app used on Debian and I've tried many. What format is being used here? Looked for an option to just store each individual note as a txt or MD but that does not seem possible either.

**Comment by ippocratis** [comment]
>  However, I don't seem to be able to view anything regardless of the note program or app used on Debian and I've tried many. 

Exports and database exposure to a user defined location serve as backup strategies not multiplatform note usage

Syncthing is just one example of sync local files to a remote you could use anything to do that e.g. copy your notally backup to a flashdrive

**Comment by PhilKes** [comment]
The notes are stored in a standard Android Room database, which is essentially a SQLite DB File.
As @ippocratis stated moving the internal database in an externally available path allows to sync the database file to multiple Android devices. 

<img src="https://github.com/user-attachments/assets/99ba3c51-7c4c-437b-bbc0-914c1baba1e0" width="300" />
Note: the other 2 files are just metadata for the database file

Obviously that means you cant really use the database file outside of the NotallyX app. Storing app data in a Room database is the default approach for android apps. Moving to a file-based storage of the notes, would be neat, but it would be a huge effort

**Comment by jawsh207** [comment]
> The notes are stored in a standard Android Room database, which is essentially a SQLite DB File. As @ippocratis stated, moving the internal database in an externally available path allows to sync the database file to multiple Android devices.
> 
> <img alt="" width="300" src="https://private-user-images.githubusercontent.com/39240633/397015257-99ba3c51-7c4c-437b-bbc0-914c1baba1e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ1Nzg1MjksIm5iZiI6MTczNDU3ODIyOSwicGF0aCI6Ii8zOTI0MDYzMy8zOTcwMTUyNTctOTliYTNjNTEtN2M0Yy00MzdiLWJiYzAtOTE0YzFiYWJhMWUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDAzMTcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmYjQ4MWFhODZhMDVhNWVkZDA0ODM0YjY4YTcyMGMzYzBhMGEyNDUwZDg0MmQ5YTljZjE5NzEwMDI4YWE3MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TNkiRSoe_8AXMYnW5PpNJo6xiHue0MBPhUBdW_Q54JI">
> Note: the other 2 files are just metadata for the database file
> 
> Obviously that means you cant really use the database file outside of the NotallyX app, except for a SQLite browser. Storing app data in a Room database is the default approach for android apps. Moving to a file-based storage of the notes, would be neat, but it would be a huge effort

Thanks to you both for answering. I agree this would be neat, I'll take your word it would be a huge effort as I have no knowledge on the matter. Previously used Nextcloud to sync notes, photos, my .kdbx database and random files. Got tired of connection/sync issues after any kind of update so switched to Syncthing and Baïkal for contacts & calendar. Note syncing is proving difficult without resorting to something like Standard Notes or Joplin.

**Comment by PhilKes** [comment]
Integration with a Room database is very easy, it offers built-in APIs that are very convenient. Over the versions there were quite a few changes to the data model of notes/lists, to convert all that into a standard format like Markdown would be tricky, even though not impossible. Especially saving certain metadata like labels, colors, pinned status, folder the note is in, would need some workarounds since they are unique NotallyX fields not present in e.g. a Markdown file.

I too would love to have an app I can use on my phone and my computer (Windows/Linux), and I thought about migrating the project to Compose Multiplatform to provide a desktop application as well, but its just a lot of effort needed for a freetime project for a single developer atm

**Comment by vertigo220** [comment]
Agree this would be nice. I'd really like something that functions similar to Keep, where I can access my notes anywhere, but of course open-source/private and better functionality. I wonder if maybe it could be done via some sort of web app that could basically be a NotallyX emulator. Though I wonder how or even if multi way syncing would be managed.

**Comment by kanlukasz** [comment]
I fully agree with this. In my case, I want to manage my notes using e.g. Typora on my laptop, while keeping them in sync using Syncthing. This way, my notes are not locked into a single app, and I can easily access and edit them across different tools and devices.

Of course, I understand that it's easy to leave a comment, but much harder to actually implement the feature. Unfortunately, I work with PHP on a daily basis and don't have any experience with Kotlin — sorry about that!
